packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -183,6 +183,19 @@ class RunCommand extends RunCommandBase {
         splitCommas: true,
         hide: true,
       );
+
+    if (!FlutterVersion.instance.isStable) {
```

Comment:
Not sure why `FlutterVersion.instance` is returning null...

Reply:
In the constructor it won't be initialized yet, thanks to the magic of our ~~service locator~~ dependency injection system.

What I've previously done is always add the argument + hide, but throwToolExit if the flags are passed on stable

Reply:
Oh that's frustrating, but thanks for letting me know! :-)